src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1370,13 +1371,15 @@ internal bool FilterAndAppendDiagnostics(DiagnosticBag accumulator, IEnumerable<
                     continue;
                 }
 
+
                 var filtered = Options.FilterDiagnostic(d);
                 if (filtered == null ||
                     (!reportSuppressedDiagnostics && filtered.IsSuppressed))
                 {
                     continue;
                 }
-                else if (filtered.Severity == DiagnosticSeverity.Error)
+                else if (filtered.Severity == DiagnosticSeverity.Error &&
+                         !filtered.IsSuppressed)
```

Comment:
>!filtered.IsSuppressed [](start = 25, length = 22)

It looks like previous ```if``` takes care of checking this. Shouldn't we respect the ```reportSuppressedDiagnostics``` option?

Reply:
We are respecting the option if that option is false we won't set `hasErrors`, but we also won't add the diagnostic to the bag. In this case we don't set `hasErrors` but we do add the diagnostic to the bag.

In other words, `hasErrors` is used to mean "stop compilation" if true, but that's not what was happening. In this case there were errors, but that doesn't imply we should stop compilation.

And reporting is just a completely different option that controls whether or not the diagnostic is output to the log at all.